bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -290,14 +299,19 @@ public WaitResult handleWaitAny(List<FutureValue> futures) throws Throwable {
                         ctx.unLock();
                         throw future.panic;
                     }
-
+                    if (future.hasWaited()) {
```

Comment:
Should we move this logic to the future.getResult method? So we do not need to check to wait in multiple places.
What happens if somebody passes already waited for the future to interop method and call get a result.

ex.
    public Object getResult() {
        if(waited){
            return createWaitOnSameFutureError();
        }
        waited = true
       return this.result;
    }

Reply:
I think we discussed this on call